contracts/GuardiansRegistration.sol

```diff
@@ -23,10 +23,43 @@ contract GuardiansRegistration is IGuardiansRegistration, ContractRegistryAccess
 		uint256 lastUpdateTime;
 	}
 	mapping (address => Guardian) public guardians;
-	mapping (address => address) public orbsAddressToEthereumAddress;
+	mapping (address => address) public orbsAddressToGuardianAddress;
 	mapping (bytes4 => address) public ipToGuardian;
 	mapping (address => mapping(string => string)) public guardianMetadata;
 
+	constructor(IGuardiansRegistration previousContract, address[] memory guardiansToMigrate) public {
+		require(previousContract != IGuardiansRegistration(0) || guardiansToMigrate.length == 0, "A guardian address list was provided for migration without the previous contract");
+
+		bytes4 ip;
+		address orbsAddr;
+		string memory name;
+		string memory website;
+		string memory contact;
+		uint registrationTime;
+		uint lastUpdateTime;
+		string memory rewardsFreqMetadata;
+
+		for (uint i = 0; i < guardiansToMigrate.length; i++) {
+			(ip, orbsAddr, name, website, contact, registrationTime, lastUpdateTime) = previousContract.getGuardianData(guardiansToMigrate[i]);
```

Comment:
may move the update of a single guardian to a function to have more readable code:
for (uint i = 0; i < guardiansToMigrate.length; i++) {
   migrateGuardianData(previousContract, guardinaAddress)
   migrateGuardianMetaData(previousContract, guardianAddress, "REWARDS_FREQUENCY_SEC")
}